### PR TITLE
🧹 Remove MultiJSON, JSON and Yajl (Use Oj instead)

### PR DESCRIPTION
**Pull request by philippthun**
- Use `Oj.load` and `Oj.dump`
- `Oj.load` => `rescue StandardError`
- `Oj.dump` + `RestController` => `mode: :compat`
- No `pretty` option anymore
- Add Cop that detects the usage of other JSON libraries than Oj

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

**Comment by philippthun** [comment]
CATS have been successfully executed (same configuration as `asha` and `scar` environments).

**Review by svkrieger — COMMENTED** [review]
Overall a very nice change which tidies up our codebase. 

Sometimes you removed `pretty` and added `mode: :compat`, sometimes not. Is this on purpose? I added comments to some places where I saw that `mode` is not specified. Later I saw this is in many more places, so probably not an issue if it's not needed.

Very cool that you added a Rubocop rule!

File: lib/cloud_controller/diagnostics.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def collect(output_directory)
       FileUtils.mkdir_p(output_directory)
 
       output_file = File.join(output_directory, output_file_name)
-      File.write(output_file, MultiJson.dump(data, pretty: true))
+      File.write(output_file, Oj.dump(data))
```
[the remark this answers is not in the document]
Comment (svkrieger) [reply lib/cloud_controller/diagnostics.rb:12]: Just saw your comment above that `compat` should only be used in Controllers.

File: spec/api/documentation/apps_api_spec.rb [hunk 2]
```diff
@@ -140,7 +140,7 @@ def after_standard_model_delete(guid)
       example 'Creating an App' do
         space_guid = VCAP::CloudController::Space.make.guid
         ports      = [1024, 2000]
-        client.post '/v2/apps', MultiJson.dump(required_fields.merge(space_guid: space_guid, diego: true, ports: ports), pretty: true), headers
+        client.post '/v2/apps', Oj.dump(required_fields.merge(space_guid: space_guid, diego: true, ports: ports)), headers
```
[the remark this answers is not in the document]
Comment (svkrieger) [reply spec/api/documentation/apps_api_spec.rb:143]: Just saw your comment above that `compat` should only be used in Controllers.

**Review by moleske — APPROVED** [review]
I will admit I did not look at every individual file.  Very excited to remove a dependency from the gemfile!

File: app/controllers/v3/root_controller.rb [hunk 3]
```diff
@@ -39,7 +39,7 @@ def v3_root
     links.merge!(create_link(:tasks))
     links.merge!(create_link(:users))
 
-    render :ok, json: MultiJson.dump({ links: }, pretty: true)
+    render :ok, json: Oj.dump({ links: })
```
[the remark this answers is not in the document]
Comment (philippthun) [reply app/controllers/v3/root_controller.rb:42]: I've only added it for v2 controllers.

**Pull request merged**